provider/README.md

```diff
@@ -0,0 +1,65 @@
+# Azure Provider
```

Comment:
What is the point of checking this "provider" in? Or maybe my question is why is this directory called _provider_ and not _modules_?

Reply:
We used to separate the provider in case it might be updated in the future. Again this based on best practices we are following but it's a design preference.

Reply:
> We used to separate the provider in case it might be updated in the future. 

What do you mean? What is the advantage of what you are doing opposed to specifying the used providers in say the `main.tf` of your module? Something like https://github.com/jenkins-x/terraform-google-jx/blob/master/main.tf.
You can still update the provider version in the latter, right? It seems the main thing you are doing in this directory is setting the providers and their version:

```
provider "azurerm" {
  version = "~>1.40.0"
}

provider "null" {
  version = "~>2.1.2"
}

provider "azuread" {
  version = "~>0.7.0"
}
```

> Again this based on best practices we are following but it's a design preference.

Do you have a pointer to these best practices? Maybe there are things we should address in the Terraform modules for the other cloud providers.

Reply:
Maybe I am just confused about the README here.